src/Compilers/Core/Portable/SourceGeneration/Nodes/NodeStateTable.cs

```diff
@@ -374,17 +373,26 @@ public bool TryModifyEntries(ImmutableArray<T> outputs, IEqualityComparer<T> com
             public void AddEntry(T value, EntryState state, TimeSpan elapsedTime, ImmutableArray<(IncrementalGeneratorRunStep InputStep, int OutputIndex)> stepInputs, EntryState overallInputState)
             {
                 _states.Add(new TableEntry(OneOrMany.Create(value), state));
+                _insertedCount += state == EntryState.Added ? 1 : 0;
```

Comment:
Presumably the builder is never multithreaded? Should we add a note mentioning it must not be?

Reply:
Yeah, the builder is absolutely not thread safe. I can add a note on the class summary, but lots of other things would break too.